be/src/exec/json_scanner.cpp

```diff
@@ -0,0 +1,485 @@
+
+    //parse jsonpath
+    if (!jsonpath.empty()) {
+        if (!_jsonPathDoc.Parse(jsonpath.c_str()).HasParseError()) {
+            if (!_jsonPathDoc.HasMember("jsonpath") || !_jsonPathDoc["jsonpath"].IsArray()) {
+                _parseJsonPathFlag = -1;// failed, has none object
+            } else {
+                _parseJsonPathFlag = 1;// success
+            }
+        } else {
+            _parseJsonPathFlag = -1;// parse failed
+        }
+    } else if (!jsonpath_file.empty()) {
+        //Read jsonpath from file, has format: file_id:md5
+        _parseJsonPathFlag = parseJsonPathFromFile(fileMgr, jsonpath_file);
+    } else {
+        _parseJsonPathFlag = 0;
+    }
+}
+
+JsonReader::~JsonReader() {
+    close();
+}
+
+void JsonReader::close() {
+    if (_closed) {
+        return;
+    }
+    if (typeid(*_file_reader) == typeid(doris::BrokerReader) || typeid(*_file_reader) == typeid(doris::LocalFileReader)) {
+        _file_reader->close();
+        delete _file_reader;
+    }
+    _closed = true;
+}
+
+int JsonReader::parseJsonPathFromFile(SmallFileMgr *smallFileMgr, std::string& fileinfo ) {
+    std::vector<std::string> parts = strings::Split(fileinfo, ":", strings::SkipWhitespace());
+    if (parts.size() != 2) {
+        LOG(WARNING)<< "parseJsonPathFromFile Invalid fileinfo: " << fileinfo;
+        return -1;
+    }
+    int64_t file_id = std::stol(parts[0]);
+    std::string file_path;
+    Status st = smallFileMgr->get_file(file_id, parts[1], &file_path);
+    if (!st.ok()) {
+        return -1;
+    }
+    std::unique_ptr<RandomAccessFile> jsonPathFile;
+    st = Env::Default()->new_random_access_file(file_path, &jsonPathFile);
+    if (!st.ok()) {
+        return -1;
+    }
+    uint64_t size = 0;
+    jsonPathFile->size(&size);
+    if (size == 0) {
+        return 0;
+    }
+    boost::scoped_array<char> pBuf(new char[size]);
+    Slice slice(pBuf.get(), size);
+    st = jsonPathFile->read_at(0, slice);
+    if  (!st.ok()) {
+        return -1;
+    }
+
+    if (!_jsonPathDoc.Parse(slice.get_data()).HasParseError()) {
+        if (!_jsonPathDoc.HasMember("jsonpath") || !_jsonPathDoc["jsonpath"].IsArray()) {
+            return -1;//failed, has none object
+        } else {
+            return 1;// success
+        }
+    } else {
+        return -1;// parse failed
+    }
+}
+
+Status JsonReader::parseJsonDoc(bool *eof) {
+    // read all, must be delete json_str
+    uint8_t* json_str = nullptr; //
+    size_t length = 0;
+    RETURN_IF_ERROR(_file_reader->read(&json_str, &length));
+    if (length == 0) {
+        *eof = true;
+        return Status::OK();
+    }
+    //  parse jsondata to JsonDoc
+    if (_jsonDoc.Parse((char*)json_str, length).HasParseError()) {
+        delete[] json_str;
+        return Status::InternalError("Parse json data for JsonDoc is failed.");
+    }
+    delete[] json_str;
+    return Status::OK();
+}
+
+size_t JsonReader::getDataByJsonPath() {
```

Comment:
In c language code style, you should define this method as:
```
Status JsonReader::get_data_by_jsonpath(size_t* max_lines) {

}
```

Reply:
ok